### PR TITLE
Various doc improvements

### PR DESCRIPTION
**Pull request by axelson**
Various doc improvements, mainly in the main `NewRelic` docs.

Changes like:
- Adding hyperlinks to functions
- Add doc to `NewRelic.create_distributed_trace_payload` to avoid warning about linking to private module `NewRelic.DistributedTrace`
- Use admonitions for warnings
- Expand the `NewRelic.set_span/2` docs to make them (hopefully) more clear

Also updates doc-related packages to latest versions.

### CODE REVIEW

**Comment by axelson** [comment]
I'm not sure why there's two CLA actions they both point to the same CLA which I have signed.

File: lib/new_relic.ex [hunk 1]
```diff
@@ -59,23 +59,26 @@ defmodule NewRelic do
...
-  Examples:
+  ## Examples
 
   ```elixir
   NewRelic.start_transaction("GenStage", "MyConsumer/EventType")
   NewRelic.start_transaction("Task", "TaskName")
   ```
 
-  **Notes:**
+  > #### Warning {: .error}
```
Comment (binaryseed) [inline lib/new_relic.ex:69]:
```suggestion
  > #### Warning
```

Comment (binaryseed) [reply lib/new_relic.ex:69]: What is this?

Comment (binaryseed) [reply lib/new_relic.ex:69]: Ok, looked it up. It's an `ex_doc` annotation. Neat

**Review by binaryseed — APPROVED** [review]
Thanks!

**Comment by axelson** [comment]
Thank you!

**Pull request merged**